tests/rules/DoctrineKeyValueStyleRuleTest.php

```diff
@@ -92,7 +93,7 @@ public function testIntegerRanges(): void
             ],
         ];
 
-        QueryReflection::getRuntimeConfiguration()->enableDoctrineKeyValueIntegerRangeChecks(true);
+        QueryReflection::getRuntimeConfiguration()->parameterTypeValidation(RuntimeConfiguration::VALIDATION_MODE_STRICT);
```

Comment:
Please create a separate test class for strict tests

Reply:
to be more precise: we need to test the integer ranges in strict and in lax mode and assert the expectations

Reply:
In that case, it seems like a single test class would help avoid a lot of code duplication. For example, I could have one test method (parameterized by the config option) that performs checks that don't change, and then another method where the assertions change with the config option.

So long as we are properly resetting the config option in the `tearDown` method, I don't think there should be any issues with the global state changing in a test method. Up to you though - if you're willing to entertain an attempt.

Reply:
I think we can just analyze the `doctrine-key-value-style-integer-ranges.php` file twice.

once from each test-class and using different runtimeconfig in each test-class

Reply:
> that we get 2 expectation errors and get reminded we need to assert lax and strict variants

By parameterizing the test, you would get 2 expectation errors, but then you only need to change one line of code. E.g.

```php
/**
 * @testWith ['lax']
 *           ['strict']
 */
public function testErrors(string $validationMode)
{
    QueryReflection::getRuntimeConfiguration()->parameterTypeValidation($validationMode);

    $expectedErrors = [...];

    $this->analyse(...);
}
```

This spares us having two exact copies of the large `$expectedErrors` array, and makes maintaining the tests easier.

(If you don't like the `@testWith` annotation, we could use a data provider method. In PHPUnit 10, the annotations are replaced with much nicer attributes, but I don't know if you're interested in updating to PHPUnit 10 any time soon.)